### PR TITLE
(DOCSP-27237): Template App page updates

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Note: We shouldn't merge this PR until the PR that updates the READMEs in the template apps repository gets merged (PR [#126](https://github.com/mongodb-university/realm-template-apps/pull/126))

### Jira

- https://jira.mongodb.org/browse/DOCSP-27237

### Staged Changes

- [Template Apps](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-27237/reference/template-apps/)

### Reminder Checklist

You might need to also update some corresponding pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-realm update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/reference/template-apps.txt [hunk 1]
```diff
@@ -78,34 +90,117 @@ option is most convenient for you.
...
+
+      Follow the on-screen instructions to get the client code using one 
+      of these methods. 
+
+      When you use this method to get the client code, the client code is not 
+      pre-populated with your App Services App ID. To use one of these
+      clients, you must :ref:`find your App ID <find-your-app-id>` to use 
+      in the client. Follow the :guilabel:`Configuration` and 
+      :guilabel:`Download the Client as a Zip File` instructions in the client 
+      :file:`README.md` to learn where to insert your App ID.
```
Comment (krollins-mdb) [inline source/reference/template-apps.txt:126]: [non-blocking, but substantial] I stumbled when I got to this sentence. My first thought was that this was meant for one of the list items above. Then I thought "this method" meant using the App Services UI. Then, I wondered if it was referring to "Download the client when you create the template app".

It _seems_ to be associated with the tab-level context. As such, I think we could improve the flow (and reduce duplication) by moving this info to the top of the UI and Realm CLI tabs. So, for Realm CLI, above "Download the client when you create the template app" we could say something like this:
> When you download client code through the App Services UI, the client code is pre-populated with your App Services App ID.

For the App Services UI tab, we could move this whole paragraph up to the top of the tab.

**Review by krollins-mdb — COMMENTED** [review]
Nice work, @dacharyc! I have a number of small suggestions and one larger suggestion about an ambiguous repeated structure.

File: source/reference/template-apps.txt [hunk 2]
```diff
@@ -78,34 +90,109 @@ option is most convenient for you.
...
-----------------------
+Get a Template App Client
+-------------------------
 
-When you create a template app, the UI includes a
-:guilabel:`Welcome to Your Application` section to help you get started
-with services and front-end code. You can:
+Some of the template apps come with working clients to explore and build 
+on when creating your applications. You can access these clients in a few
+ways:
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/reference/template-apps.txt:100]: Ah, good point!

File: source/reference/template-apps.txt [hunk 3]
```diff
@@ -78,34 +90,109 @@ option is most convenient for you.
...
+      **Download the client when you create the template app**
 
-2. Follow on-screen guides to customize your app.
+      When you create a template app using the App Services UI, the UI 
+      prompts you with two options to get the client code immediately 
+      after creating the template:
 
-3. Use the {+cli-bin+} to download the code for the template app so you can build the front end
-   (if one is included) and interact with the app through the CLI:
+      - Download the client code in your preferred programming language
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/reference/template-apps.txt:113]: I reworded this to "your preferred client" - thoughts?

File: source/reference/template-apps.txt (hunk 1 above)
Comment (dacharyc) [reply source/reference/template-apps.txt:126]: Makes sense!

**Review by krollins-mdb — APPROVED** [review]
@dacharyc, this looks great! Thank you for the excellent work on this. 🚢

**Pull request merged**